src/WebJobs.Script/Host/ScriptHost.cs

```diff
@@ -461,12 +462,6 @@ private void InitializeFileSystem()
             if (!_settingsManager.FileSystemIsReadOnly)
             {
                 FileUtility.EnsureDirectoryExists(ScriptConfig.RootScriptPath);
-
-                if (!File.Exists(_hostConfigFilePath))
```

Comment:
Because this code was in place, even previously if the file were missing, we'd default the config. The issue comes in for read-only file systems where we can't write this file, and my changes here fix that. I could leave code here, but I lean towards deleting it now.